src/camera.py

```diff
@@ -0,0 +1,212 @@
+        0 refers to frontfacing while 150 refers to 60 degrees past vertical
+        rotating upward. 
+
+        :param degree: The degree to set the tilt servo to.
+        :type degree: float
+        """
+        self._current_tilt_angle = self.validate_degree(degree,
+                                                        self.MIN_DEGREE,
+                                                        self.MAX_TILT_DEGREE)
+        self._tilt_servo.write(self.validate_degree(self._current_tilt_angle +
+                                                    self._tilt_offset,
+                                                    self.MIN_DEGREE,
+                                                    self.MAX_TILT_DEGREE))
+
+
+    def calibrate_tilt(self, degree):
+        """
+        Calibrates camera tilt.
+
+        Controls offset in [-180, 180] degrees for tilt. Writes new offset value to
+        'SunFounder_PiCar-V/remote_control/remote_control/driver/config'
+        to 'tilt_offset'.
+
+        :param degree: The amount of degrees to add or subtract for calibration.
+        :type degree: float
+        """
+        self._tilt_offset = self.validate_degree(degree, self.MIN_CALI_DEGREE,
+                                                         self.MAX_CALI_DEGREE)
+        self._DB.set('tilt_offset', self._tilt_offset)
+
+
+    def get_tilt(self):
+        """
+        :return: The current tilt angle (in degrees).
+        :rtype: float
+        """
+        return self._current_tilt_angle
+
+
+    def validate_degree(self, degree, min_degree, max_degree):
+        """
+        Takes input and restricts to specifcied min and max.
+
+        Returns the value in the degree parameter that is bounded by the 
+        min_degree and max_degree parameters.
+
+        :param degree: The value to be bounded between min and max degree.
+        :type degree: float
+
+        :param min_degree: The lower bound of the validation.
+        :type min_degree: float
+
+        :param max_degree: the upper bound of the validation.
+        :type max_degree: float
+
+        :return: The value of degree bounded between the min and max parameters.
+        :rtype: float
+        """
+        return max(min(max_degree, degree), min_degree)
+
+
+    def test_pan(self):
+        """
+        Tests the panning of the camera
+
+        sets the camera to 90 degrees then to right most value at 0 degrees. 
+        Then pans the camera from right to left giving over the maximum
+        values showing validation function working as intended, ending at 180 
+        degrees. Then pans the camera from left to right ending at 0 degrees.
+        resets the camera to 90 degrees.
+        """
+        self.set_pan(90)
+        time.sleep(.2)
+        self.set_pan(0)
+        time.sleep(.2)
+        print ("panning from right to left")
+        for i in range (-20, 200, 2):
+            self.set_pan(i)
+            time.sleep(.04)
```

Comment:
Can we keep all the sleep times the same? also maybe move it into a variable so it's easy to change.

Reply:
because of the distance the camera has to move it needs a little more leeway for the initial turn rather than the looped panning. I could rename them to long_sleep and short_sleep, These values will only be used in the testing.